server/game/cycle/stichGranter.js

```diff
@@ -2,98 +2,54 @@ import {GameMode} from '../../../shared/game/gameMode';
 
 const StichGranter = {
     determineWinner: function (mode, trumpfColor, playedCards, players) {
-        let winnerIndex = 0;
-        let winnerCard;
+        const firstPlayedColor = playedCards[0].color;
+        const isModeTrumpf = mode === GameMode.TRUMPF;
+        const trumpfAndFirstPlayedColorCards = playedCards.map(addIndexAndAdjustTrumpfValues)
+                .filter((playedCard) => playedCard.color === firstPlayedColor
+                || (isModeTrumpf && isTrumpf(playedCard)));
+
+        switch (mode) {
```

Comment:
We may write this as simple if instead of a switch.
if (mode === GameMode.UNDEUFE) {
} else {
}
should do exactly the same, doesn't it?